packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/MutableRealmIntInternal.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.internal
+
+import io.realm.kotlin.internal.RealmObjectHelper.NOT_IN_A_TRANSACTION_MSG
+import io.realm.kotlin.internal.interop.PropertyKey
+import io.realm.kotlin.internal.interop.RealmInterop
+import io.realm.kotlin.types.BaseRealmObject
+import io.realm.kotlin.types.MutableRealmInt
+
+internal class ManagedMutableRealmInt(
+    private val obj: RealmObjectReference<out BaseRealmObject>,
+    private val propertyKey: PropertyKey,
+    private val converter: RealmValueConverter<Long>
+) : MutableRealmInt() {
+
+    private enum class Operation {
```

Comment:
Seems like a very cumbersome way of aligning two different implementation paths just to split them in the `operationInternal` ... why?

Reply:
But you are introducing an enum just for the sake of being able to `when`-it. Then maybe do a inline method that takes a lambda with the different inner operations and the message.